source/common/event/dispatcher_impl.cc

```diff
@@ -41,6 +41,14 @@ DispatcherImpl::DispatcherImpl(Buffer::WatermarkFactoryPtr&& factory, Api::Api&
 
 DispatcherImpl::~DispatcherImpl() {}
 
+void DispatcherImpl::initializeStats(Stats::Scope& parent_scope) {
+  std::string thread_id = api_.threadFactory().currentThreadId()->debugString();
```

Comment:
Prefer not to use a debug string interface, because this is subject to change, i.e. there are no long term guarantees on debug representations.

Reply:
The `ThreadId` class doesn't really offer anything better currently. I could do a bit of additional spelunking and add a `int64_t id() const` method there, but I also wonder if the underlying platform-specific thread IDs are hidden from the interface on purpose.

Reply:
It is hidden on purpose. How do you think we can provide a stable ID given that debug is not guaranteed to be that?

Reply:
Could we just assign workers an ordinal ID, e.g.` [0, 1, 2, ..N-1]` for N workers and use that? The advantage is this will be consistent across platforms.

Reply:
A trivial way would be to just assign sequential IDs ("worker 1", "worker 2"...), which seems fine to me. But then you can't correlate anything going awry in these stats with anything you might see in the logs, which are tagged with actual thread IDs...

Reply:
What if we have thread's log their known name when they startup? That way, you can do the join, albeit with some additional work?

Reply:
Opened #6548 after chatting with @htuch.

Reply:
... and closed #6548 again after chatting with @jmarantz. I'm just doing to identify worker dispatchers with a sequential ID as discussed above, and we can worry about how to map those to thread IDs in a separate PR.

Reply:
Ok, per @htuch, I now log the thread ID for debugging (in `run`, below). Note that I can't do it here because `initializeStats` isn't necessarily called from the same thread as the dispatcher's event loop.